src/rdkafka_metadata.c

```diff
@@ -1233,9 +1233,11 @@ static void rd_kafka_metadata_leader_query_tmr_cb (rd_kafka_timers_t *rkts,
                     rtmr->rtmr_interval * 2 / 1000 >=
                     rk->rk_conf.metadata_refresh_interval_ms)
                         rd_kafka_timer_stop(rkts, rtmr, 1/*lock*/);
-                else
+                else {
+                        rtmr->rtmr_interval = rtmr->rtmr_interval * 2;
```

Comment:
Sorry for missing this the last review round, but I think a better way is pass the interval to backoff:
`..backoff(rkts, rtmr, (int)rtmr->rtmr_interval * 2)`

(and then remove the added { .. } since they're no longer needed.

Reply:
Will that work? Won't it end being the same problem were the backoff is only rtmr->rtmr_interval * 2 always? I don't see the rtmr_interval getting updated in rd_kafka_timer_backoff.

Reply:
You're right, what I think we'll want to do is:
 1. rename the function to rd_kafka_exp_backoff()
 2.  remove the backoff_us argument
 3. in exp_backoff() if the timer is scheduled update rtmr_interval as you suggested, e.g., something like:
```c
void rd_kafka_timer_exp_backoff (rd_kafka_timers_t *rkts, rd_kafka_timer_t *rtmr {
	rd_kafka_timers_lock(rkts);
	if (rd_kafka_timer_scheduled(rtmr)) {
                rtmr->rtmr_interval *= 2;
		rd_kafka_timer_unschedule(rkts, rtmr);
        }
	rd_kafka_timer_schedule(rkts, rtmr, rtmr->rtmr_interval);
	rd_kafka_timers_unlock(rkts);
}
```

Reply:
Sounds good. I addressed the comments. Thanks!